code/modules/power/singularity/emitter.dm

```diff
@@ -435,33 +457,32 @@
 	button_icon_state = "mech_zoom_on"
 
 /datum/action/innate/protoemitter/firing/Activate()
-	if(PE.manual)
-		playsound(PE,'sound/mecha/mechmove01.ogg', 50, TRUE)
-		PE.manual = FALSE
+	if(proto_emitter.manual)
+		playsound(proto_emitter,'sound/mecha/mechmove01.ogg', 50, TRUE)
+		proto_emitter.manual = FALSE
 		name = "Switch to Manual Firing"
 		desc = "The emitter will only fire on your command and at your designated target"
 		button_icon_state = "mech_zoom_on"
-		for(var/obj/item/I in U.held_items)
-			if(istype(I, /obj/item/turret_control))
-				qdel(I)
+		for(var/obj/item/item in buckled_mob.held_items)
+			if(istype(item, /obj/item/turret_control))
+				qdel(item)
 		UpdateButtonIcon()
 		return
-	else
-		playsound(PE,'sound/mecha/mechmove01.ogg', 50, TRUE)
-		name = "Switch to Automatic Firing"
-		desc = "Emitters will switch to periodic firing at your last target"
-		button_icon_state = "mech_zoom_off"
-		PE.manual = TRUE
-		for(var/V in U.held_items)
-			var/obj/item/I = V
-			if(istype(I))
-				if(U.dropItemToGround(I))
-					var/obj/item/turret_control/TC = new /obj/item/turret_control()
-					U.put_in_hands(TC)
-			else //Entries in the list should only ever be items or null, so if it's not an item, we can assume it's an empty hand
-				var/obj/item/turret_control/TC = new /obj/item/turret_control()
-				U.put_in_hands(TC)
-		UpdateButtonIcon()
+	playsound(proto_emitter,'sound/mecha/mechmove01.ogg', 50, TRUE)
+	name = "Switch to Automatic Firing"
+	desc = "Emitters will switch to periodic firing at your last target"
+	button_icon_state = "mech_zoom_off"
+	proto_emitter.manual = TRUE
+	for(var/things in buckled_mob.held_items)
+		var/obj/item/item = things
+		if(istype(item))
+			if(buckled_mob.dropItemToGround(item))
+				var/obj/item/turret_control/turret_control = new /obj/item/turret_control()
+				buckled_mob.put_in_hands(turret_control)
+		else //Entries in the list should only ever be items or null, so if it's not an item, we can assume it's an empty hand
+			var/obj/item/turret_control/turret_control = new /obj/item/turret_control()
+			buckled_mob.put_in_hands(turret_control)
```

Comment:
Could you use continues to make this less nested?

Reply:
aight